### PR TITLE
Fix component events matrix in docs

### PR DESCRIPTION
**Pull request by aliabd**
Components events matrix in the docs is broken, this PR fixes it and adds the missing events too. 

Closes one issue in #4804:
- [x] Components events matrix not showing any event

Closes: #4104

### CODE REVIEW

**Comment by freddyaboulton** [comment]
Also closes #4104

**Comment by freddyaboulton** [comment]
Nice @aliabd ! How do I test this? It's not showing up correctly for me in the preview

image

**Comment by aliabd** [comment]
@freddyaboulton yup missed that issue! 

I'm not sure why it isn't showing up correctly either, debugging now.

**Comment by aliabd** [comment]
@freddyaboulton just fixed, please take another look :)

File: js/_website/generate_jsons/src/docs/__init__.py [hunk 1]
```diff
@@ -54,35 +54,38 @@ def add_demos():
...
-    "Play()",
-    "Pause()",
-    "Stream()",
-    "Blur()",
-    "Upload()",
-]
-
-
-def add_supported_events():
+events = [
```
Comment (freddyaboulton) [inline js/_website/generate_jsons/src/docs/__init__.py:57]: How can we keep this up to date in the future?

Maybe we can programmatically import all `EventListeners` and programmatically document all the properties that are of instance `EventListenerMethod`. Or we have `EventListenerMethod` update a global dict with the event name?

**Comment by aliabd** [comment]
thanks @freddyaboulton was thinking the same as i was making the PR. I'll go through and refactor a lot of this soon.

File: js/_website/generate_jsons/src/docs/__init__.py (hunk 1 above)
Comment (aliabd) [reply js/_website/generate_jsons/src/docs/__init__.py:57]: so I actually got stuck on this earlier: when I list the attributes of one of the EventListeners classes, it doesn't show any of the EventListenerMethods.. not sure why.

```
<class 'gradio.events.Changeable'>
{'__module__': 'gradio.events', '__init__': <function Changeable.__init__ at 0x7fcf3004f1f0>, '__doc__': None}
```

Comment (aliabd) [reply js/_website/generate_jsons/src/docs/__init__.py]: just fixed it @freddyaboulton, not hardcoded anymore. can you take another look? 

Btw what is 'end'? not showing up as a method for any component

File: js/_website/generate_jsons/src/docs/__init__.py [hunk 2]
```diff
@@ -54,35 +55,26 @@ def add_demos():
...
-]
-
-
-def add_supported_events():
+def create_events_matrix():
+    events = []
+    for c in EventListener.__subclasses__():
+        methods = c().__dict__
+        for m in methods: 
+            if m[:1] != '_' and isinstance(methods[m], EventListenerMethod) and m[:1] not in events:
```
Comment (freddyaboulton) [inline js/_website/generate_jsons/src/docs/__init__.py:63]: I think you mean `m not in events` here?

Comment (freddyaboulton) [reply js/_website/generate_jsons/src/docs/__init__.py:63]: Otherwise looks good!

Comment (aliabd) [reply js/_website/generate_jsons/src/docs/__init__.py:63]: lol

File: js/_website/generate_jsons/src/docs/__init__.py (hunk 1 above)
Comment (freddyaboulton) [reply js/_website/generate_jsons/src/docs/__init__.py]: The end event should be triggered when the video/audio stops playing. We should add it to document decorator for Playable class. I think that was an oversight.

Comment (aliabd) [reply js/_website/generate_jsons/src/docs/__init__.py]: got it thanks! added it, and showing up correctly with video and audio :) will merge. thanks for the help!

**Pull request merged**